### DISCUSSION TITLE
Split .env loading and process.env update

### DISCUSSION THREAD

**Discussion by djfo** [motdotla/dotenv#663] [Ideas]
Hi everybody,

this is the first time I use the *Discussions* feature and also the first time I'm trying to participate in the development of the `dotenv` package.

I have a suggestion related to the interface/API of the package. 

The package exports the functions `parse` and `config`, where `parse` parses the contents of a `.env` file and `config` updates `process.env` based on the key-value pairs read from the `.env` file.

Since `config` is implemented using `parse`, I hoped to also find a function that parses the `.env` file (taking a configuration as an argument like `config`) and returns a `DotenvParseOutput`, without updating `process.env`. Or alternatively, a function that loads the contents of `.env` (according to the configuration) and returns a string or buffer (essentially something that can be passed to `parse` in turn).

However, there is no such function.

I suggest to add this function, maybe named `load`, with the same interface as `config`, but not updating `process.env`:

```typescript
export function load(options?: DotenvConfigOptions): DotenvConfigOutput;
```

It is easily implemented in code using the `dotenv` package, but I think it would make a nice addition to the package.

Rationale: I expect that a simple call to `config` is the main use case for this package. However, sometimes side-effect-free loading of `.env` (in the sense of only loading the key-value pairs without updating the environment) may be desirable.

Please let me know if such a change is welcome, I'd be happy to open a corresponding PR if so.

**Comment by motdotla** [comment]
@djfo this is a great suggestion. Thanks @djfo. Others have mentioned this as well.

It's something we will try to work on soon.

**Reply by djfo** [reply]
@motdotla I would like to open a PR if that's ok?

**Reply by motdotla** [reply]
Yes, please do. I will try to get to it in a reasonable time.

**Reply by motdotla** [reply]
@djfo this has been added to `16.1.0-rc2` and will soon be released as `16.1.0`. See the `populate` function.

**Reply by motdotla** [reply]
here are docs: https://github.com/motdotla/dotenv#populate
